src/highlight-support.js

```diff
@@ -28,12 +27,29 @@ const highlightSupport = {
     }
   },
 
-  highlightRange (editableHost, highlightId, startIndex, endIndex, dispatcher, type = 'comment') {
+
+  // This function was changed to track matches when text is added to the start of a component, but multiple white spaces break it in a strict sense
+  // The function works in the editor and in browsers, but tests with multiple white spaces will fail.
+  // Browsers change the white spaces to &nbsp and the function works, and the tests in highlight.spec.js have been updated to represent this.
+
+  highlightRange (editableHost, text, highlightId, startIndex, endIndex, dispatcher, type = 'comment') {
     if (this.hasHighlight(editableHost, highlightId)) {
       this.removeHighlight(editableHost, highlightId)
     }
+    const blockText = highlightText.extractText(editableHost)
```

Comment:
You can move this find match and move the index behavior to a own function so it is easier to understand.

Reply:
You can also add a note to this function that you will find only matches which are bigger than the given index.

Reply:
Nice one, I've updated it, let me know what you think